contracts/adapters/dp/MakerVault.sol

```diff
@@ -361,4 +323,60 @@ contract MakerVaultAdapter is IDebtPosition, FactoryFriendly {
 
         return result;
     }
+
+    /// @notice calculates collateral value in the vault, priced in base asset
+    /// @dev formula is Vat.urn.ink * Vat.ilk.spot * Spot.ilk.mat
+    /// @return collateralValue as fixed point ray (27 decimals)
+    function collateralValue() public view returns (uint256) {
+        // get Ilk (collateral type)
+        // rate -> stablecoin debt multiplier (accumulated stability fees)
+        // spot -> collateral price with safety margin, i.e. the maximum stablecoin allowed per unit of collateral
+        (, , uint256 spot, , ) = IVat(vat).ilks(ilk);
+
+        // get Urn (collateral deposit)
+        // ink -> collateral balance
+        (uint256 ink, ) = IVat(vat).urns(ilk, urnHandler);
+
+        // Get Spot (price oracle)
+        // mat -> the liquidation ratio
+        (, uint256 mat) = ISpotter(spotter).ilks(ilk);
+
+        return _mul(_mul(_scaleUp(ink), spot), mat);
+    }
+
+    /// @notice calculates total outstanding debt
+    /// @dev formula is Vat.urn.art * Vat.ilk.rate
+    /// @return debtValue as ray (27 decimals)
+    function debtValue() public view returns (uint256) {
+        // get Ilk (collateral type)
+        // rate -> stablecoin debt multiplier (accumulated stability fees)
+        // spot -> collateral price with safety margin, i.e. the maximum stablecoin allowed per unit of collateral
+        (, uint256 rate, , , ) = IVat(vat).ilks(ilk);
+
+        // get Urn (collateral deposit)
+        // art -> outstanding stablecoin debt
+        (, uint256 art) = IVat(vat).urns(ilk, urnHandler);
+
+        return _mul(_scaleUp(art), rate);
+    }
+
+    /// @dev multiplies two fixed point integers in ray scale
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256) {
+        return (x * y) / RAY;
+    }
+
+    /// @dev multiplies two fixed point integers in ray scale
+    function _div(uint256 x, uint256 y) internal pure returns (uint256) {
+        return (x * RAY) / y;
+    }
+
+    /// @dev scales up a fixed point integer in wad scale to ray scale
+    function _scaleUp(uint256 _wad) internal pure returns (uint256) {
```

Comment:
Why is `_scaleUp` necessary? Aren't you always going to have to scale it back down anyway?

Reply:
Address in latest commit